### ISSUE TITLE
Support for permanently-delete in azurerm_machine_learning_workspace

### ISSUE THREAD

**Issue by domirohner** [hashicorp/terraform-provider-azurerm#22171]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure recently added a new feature soft-deletion for Azure Machine Learning Workspaces (https://learn.microsoft.com/en-us/azure/machine-learning/concept-soft-delete?view=azureml-api-2).
With this feature you need to hard-delete the ML workspace before you can re-create the workspace with the same name again (e.g. in pipelines). 

It would be really helpful if you could also integrate the permanantly-delete option (https://learn.microsoft.com/en-us/cli/azure/ml/workspace?view=azure-cli-latest#az-ml-workspace-delete) in Terraform.

### New or Affected Resource(s)/Data Source(s)

azurerm_machine_learning_workspace

### Potential Terraform Configuration

Maybe it can be solved with a similar feature as with the KeyVault (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault):
```hcl
provider "azurerm" {
  features {
    azurerm_machine_learning_workspace {
      purge_soft_delete_on_destroy = false
    }
  }
}
```

**Comment by meierale** [comment]
I was just about to open this issue as well. As we have automated jobs that create and destroy our machine learning development & test infrastructure on a daily basis, our setup is currently broken until such a feature is implemented.

**Comment by youlitai003** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Community Note
> * Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
> * Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
> * If you are interested in working on this issue or have submitted a pull request, please leave a comment
> 
> ### Description
> Azure recently added a new feature soft-deletion for Azure Machine Learning Workspaces (https://learn.microsoft.com/en-us/azure/machine-learning/concept-soft-delete?view=azureml-api-2). With this feature you need to hard-delete the ML workspace before you can re-create the workspace with the same name again (e.g. in pipelines).
> 
> It would be really helpful if you could also integrate the permanantly-delete option (https://learn.microsoft.com/en-us/cli/azure/ml/workspace?view=azure-cli-latest#az-ml-workspace-delete) in Terraform.
> 
> ### New or Affected Resource(s)/Data Source(s)
> azurerm_machine_learning_workspace
> 
> ### Potential Terraform Configuration
> Maybe it can be solved with a similar feature as with the KeyVault (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault):
> 
> ```terraform
> provider "azurerm" {
>   features {
>     azurerm_machine_learning_workspace {
>       purge_soft_delete_on_destroy = false
>     }
>   }
> }
> ```

My automation is currently broken now and either "azurerm_machine_learning_workspace" or "machine_learning_workspace" are not supported in features block.

**Comment by sharccc** [comment]
Internal reference 114726 - HashiCorp Terraform Support

**Comment by sharccc** [comment]
HashiCorp internal reference IPL-4256 (Terraform Support)

**Comment by sturlabragason-devoteam** [comment]
Until it's supported I'm using:

`
resource "null_resource" "ml_workspace_destroy" {
  provisioner "local-exec" {
    when    = destroy
    command = <<-EOT
      echo "Logging in to Azure..."
      az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID --output none
      
      echo "Obtaining the Azure access token..."
      TOKEN=$(az account get-access-token --query accessToken -o tsv)

      echo "Deleting the Azure ML Workspace using REST API..."
      RESPONSE=$(curl -s -w "\\nHTTP Status: %%{http_code}\\n" -X DELETE "https://management.azure.com/subscriptions/$ARM_SUBSCRIPTION_ID/resourceGroups/name-resources/providers/Microsoft.MachineLearningServices/workspaces/name-workspace?api-version=2023-04-01-preview&forceToPurge=true" \
           -H "Authorization: Bearer $TOKEN" \
           -H "Content-Type: application/json" \
           -d '{
                "properties": {
                    "force": true,
                    "purge": true,
                    "purgeResources": true
                }
             }')
      echo "$RESPONSE"
    EOT
  }
}
`

This is an AzDo ubuntu agent so, which the reason I'm just using the already installed az cli to get the token for the request.

Be aware that the request is async so it takes somewhere between 10-120s before the workspace is purged.

**Comment by sharccc** [comment]
@rcskosir is there any update we can share with @meierale ?

**Comment by domirohner** [comment]
Is there any update on this?

**Comment by naveensy** [comment]
is this going to be picked up?

**Comment by sharccc** [comment]
internal record closed referencing https://github.com/hashicorp/terraform-provider-azurerm/pull/25624 
merged 4 commits into [main] 5 hours ago

**Comment by stephybun** [comment]
Closed by #25624
